client/src/components/login/LoginForm.tsx

```diff
@@ -0,0 +1,193 @@
+import { useForm } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { z } from 'zod';
+import { Card } from './../ui/Card';
+import logo from '/src/assets/logo-light.png';
+import classes from './LoginForm.module.scss';
+import { useState } from 'react';
+import { useMutation } from '@tanstack/react-query';
+import { useOutsideClick } from '../../hooks';
+import { useDispatch } from 'react-redux';
+import { hideLogin, showSignUp, login as storeLogin } from '../../store';
+import { BiError } from 'react-icons/bi';
+
+const schema = z.object({
+	email: z.string().email('Invalid email'),
+	password: z.string().min(8, 'Password is too short'),
+});
+
+type LoginData = {
+	email: string;
+	password: string;
+};
+
+export const LoginForm = () => {
+	const [formError, setFormError] = useState<string | null>(null);
+	const [emailFocus, setEmailFocus] = useState(false);
+	const [passwordFocus, setPasswordFocus] = useState(false);
+	const dispatch = useDispatch();
+	const ref = useOutsideClick(() => {
+		dispatch(hideLogin());
+	});
+
+	const {
+		register,
+		handleSubmit,
+		watch,
+		formState: { errors },
+		clearErrors,
+	} = useForm<LoginData>({ resolver: zodResolver(schema) });
+
+	const login = async (credentials: { email: string; password: string }) => {
+		const response = await fetch(
+			'http://localhost:8080/auth/authenticate',
+			{
+				method: 'POST',
+				headers: {
+					'Content-Type': 'application/json',
+				},
+				body: JSON.stringify({
+					email: credentials.email,
+					password: credentials.password,
+				}),
+			}
+		);
+		if (!response.ok) {
+			if (response.status === 403) {
+				setFormError('Invalid credentials');
+				return;
+			}
+			setFormError('Unexpected error, try again later');
+			return;
+		}
```

Comment:
if (response.status === 403) {}
if (!response.ok) {}

unnecessary nesting of logic that makes code unreadable